python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,980 @@
+
+    def stackCoadd(self, dcrCoadds):
+        """Add a list of sub-band coadds together.
+
+        Parameters
+        ----------
+        dcrCoadds : list of lsst.afw.image.exposure.exposure.ExposureF
+            A list of coadd exposures, each exposure containing
+            the model for one subfilter.
+
+        Returns
+        -------
+        lsst.afw.image.exposure.exposure.ExposureF
+            A single coadd exposure that is the sum of the sub-bands.
+        """
+        coaddExposure = dcrCoadds[0].clone()
+        for coadd in dcrCoadds[1:]:
+            coaddExposure.maskedImage += coadd.maskedImage
+        return coaddExposure
+
+    def fillCoadd(self, dcrModels, skyInfo, tempExpRefList, weightList):
+        """Create a list of coadd exposures from a list of masked images.
+
+        Parameters
+        ----------
+        dcrModels : list of lsst.afw.image.maskedImageF
+            A list of masked images, each containing the model for one subfilter
+        skyInfo : lsst.skymap.discreteSkyMap.DiscreteSkyMap
+            Patch geometry information, from getSkyInfo
+        tempExpRefList : List of ButlerDataRefs
+            The data references to the input warped exposures.
+        weightList : list of floats
+            The weight to give each input exposure in the coadd
+
+        Returns
+        -------
+        list of lsst.afw.image.exposure.exposure.ExposureF
+            A list of coadd exposures, each exposure containing
+            the model for one subfilter.
+        """
+        dcrCoadds = []
+        for model in dcrModels:
+            coaddExposure = afwImage.ExposureF(skyInfo.bbox, skyInfo.wcs)
+            coaddExposure.setCalib(self.scaleZeroPoint.getCalib())
+            coaddExposure.getInfo().setCoaddInputs(self.inputRecorder.makeCoaddInputs())
+            self.assembleMetadata(coaddExposure, tempExpRefList, weightList)
+            coaddUtils.setCoaddEdgeBits(model.mask, model.variance)
+            coaddExposure.setMaskedImage(model)
+            dcrCoadds.append(coaddExposure)
+        return dcrCoadds
+
+    def convolveDcrModelPlane(self, maskedImage, dcr, bbox=None, useInverse=False):
+        """Shift a masked image.
+
+        Parameters
+        ----------
+        maskedImage : lsst.afw.image.maskedImageF
+            The input masked image to shift.
+        dcr : lsst.afw.geom.Extent2I
+            Shift calculated with `dcrShiftCalculate`.
+        bbox : lsst.afw.geom.box.Box2I, optional
+            Sub-region of the masked image to shift.
+            Shifts the entire image if None.
+        useFFT : bool, optional
+            Perform the convolution with an FFT?
+        useInverse : bool, optional
+            Use the reverse of `dcr` for the shift.
+
+        Returns
+        -------
+        lsst.afw.image.maskedImageF
+            A masked image, with the pixels within the bounding box shifted.
+
+        Raises
+        ------
+        NotImplementedError
```

Comment:
Put type in backticks.

Reply:
Not according to https://developer.lsst.io/python/numpydoc.html#raises